src/components/ScaleLattice.vue

```diff
@@ -1,19 +1,282 @@
 otonalFundamental
 <script setup lang="ts">
-import type { Scale } from "scale-workshop-core";
-import { ref } from "vue";
-
-defineProps<{
+import type { Interval, Scale } from "scale-workshop-core";
+import { computed, type ComputedRef } from "vue";
+import { PRIMES } from "xen-dev-utils";
+import { euclideanDistance, getPairCombinations } from "@/utils";
+const props = defineProps<{
   scale: Scale;
   width: number;
   height: number;
   heldScaleDegrees: Set<number>;
   keyColors: string[]; // Use key colors in visualization if you like. Not synced with scale size, though...
 }>();
 
-const canvas = ref<HTMLCanvasElement | null>(null);
+type Coord = { x: number; y: number };
+type Node = {
+  x: number;
+  y: number;
+  label: string;
+  index: number;
+  fill: string;
+};
+
+type Coords = {
+  [key: string]: Coord;
+};
+
+const coords: Coords = {
+  "1": {
+    x: 0,
+    y: 0,
+  },
+  "2": {
+    x: 0,
+    y: 0,
+  },
+  "3": {
+    x: 40,
+    y: 0,
+  },
+  "5": {
+    x: 0,
+    y: -40,
+  },
+  "7": {
+    x: 13,
+    y: -11,
+  },
+  "11": {
+    x: -14,
+    y: -18,
+  },
+  "13": {
+    x: -8,
+    y: -4,
+  },
+  "17": {
+    x: -5,
+    y: -32,
+  },
+  "19": {
+    x: 7,
+    y: -25,
+  },
+  "23": {
+    x: 20,
+    y: -6,
+  },
+};
+
+// === Computed state ===
+const canLatticePrimes = computed(() =>
+  // hardcoded for now to the 23-limit
+  props.scale.intervals.reduce(
+    (canLattice: boolean, interval: Interval) =>
+      canLattice &&
+      interval.monzo
+        .toIntegerMonzo()
+        .slice(9)
+        .reduce(
+          (isZero: boolean, component: number) => isZero && component === 0,
+          true
+        ),
+    true
+  )
+);
+
+function maybeGetPrimeEquave(equave: Interval): number | null {
+  const intmonzo: number[] = equave.monzo.toIntegerMonzo();
+  return intmonzo.reduce(
+    (
+      primeIndex: number | null,
+      component: number,
+      i: number
+    ): number | null => {
+      if (component === 1 && !primeIndex) {
+        return i;
+      }
+      if (component !== 0) {
+        console.error("Equave is not a prime number");
```

Comment:
An error in the console is too obscure if this is intended to be seen by the user.

Reply:
It was not intended to be seen by the user, but any suggestion on how to improve this is welcome.